src/pages/Aggregator/Aggregator.hooks.test.tsx

```diff
@@ -0,0 +1,342 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { act } from 'react-dom/test-utils';
+import { MockStoreEnhanced } from 'redux-mock-store';
+import { Web3Provider } from '@ethersproject/providers';
+import { Provider } from 'react-redux';
+import { ReactNode } from 'react';
+import { getStore, rootReducer } from '../../store';
+import {
+  useAggregatorDropdowns,
+  useConnectedChain,
+  useWalletAddress,
+} from './Aggregator.hooks';
+import { AggregatorInputs } from './Aggregator.fields';
+import { ChainEnum } from '../../config/chains';
+import { Reducers } from '../../constants';
+import { appActions, appReducer } from '../../store/app/app.slice';
+import {
+  aggregatorActions,
+  aggregatorReducer,
+} from '../../store/aggregator/aggregator.slice';
+import * as aggregatorSelectors from '../../store/aggregator/aggregator.selectors';
+import * as appSelectors from '../../store/app/app.selectors';
+import { switchConnectedChain } from '../../utils/switchConnectedChain';
+
+type TestStoreProviderProps = {
+  children: ReactNode;
+};
+
+const createMockStore = () => {
+  const mockRootReducer = {
+    [Reducers.Aggregator]: aggregatorReducer,
+    [Reducers.App]: appReducer,
+  };
+
+  const store = {
+    ...getStore(undefined, mockRootReducer as unknown as typeof rootReducer),
+  };
+
+  return store;
+};
+
+const createTestStoreProvider =
+  (mockedStore: MockStoreEnhanced<unknown, {}>) =>
+  ({ children }: TestStoreProviderProps) =>
+    <Provider store={mockedStore}>{children}</Provider>;
+
+const testSequence = (func: jest.Mock, ...args: AggregatorInputs[]) => {
+  args.forEach((inputType: AggregatorInputs, i) =>
+    expect(func).toHaveBeenNthCalledWith(i + 1, inputType)
+  );
+};
+
+let resetField: jest.Mock;
```

Comment:
I think you could jest settle these variables here with `jest.fn()`. Then the `jest.clearAllMocks();` function in the beforeEach block should reset all mock data before each test

https://jestjs.io/docs/jest-object#jestclearallmocks